run-make-check.sh

```diff
@@ -60,45 +78,63 @@ function run() {
         echo "This probably means distribution $ID is not supported by run-make-check.sh" >&2
     fi
 
+    if ! type ccache > /dev/null 2>&1 ; then
+        echo "ERROR: ccache could not be installed"
+        exit 1
+    fi
+    if ! type jq > /dev/null 2>&1 ; then
+        echo "WARNING: jq could not be installed: make check will probably fail."
+    fi
+
     if test -f ./install-deps.sh ; then
 	$DRY_RUN source ./install-deps.sh || return 1
     fi
 
     # Init defaults after deps are installed. get_processors() depends on coreutils nproc.
     DEFAULT_MAKEOPTS=${DEFAULT_MAKEOPTS:--j$(get_processors)}
     BUILD_MAKEOPTS=${BUILD_MAKEOPTS:-$DEFAULT_MAKEOPTS}
+    test "$BUILD_MAKEOPTS" && echo "make will run with option(s) $BUILD_MAKEOPTS"
     CHECK_MAKEOPTS=${CHECK_MAKEOPTS:-$DEFAULT_MAKEOPTS}
 
-    CMAKE_PYTHON_OPTS=
-    if ! type python2 > /dev/null 2>&1 ; then
-        CMAKE_PYTHON_OPTS="-DWITH_PYTHON2=OFF -DWITH_PYTHON3=ON -DMGR_PYTHON_VERSION=3"
+    if type python2 > /dev/null 2>&1 ; then
+        # gtest-parallel requires Python 2
+        CMAKE_PYTHON_OPTS="-DWITH_GTEST_PARALLEL=ON"
+    else
+        CMAKE_PYTHON_OPTS="-DWITH_PYTHON2=OFF -DWITH_PYTHON3=ON -DMGR_PYTHON_VERSION=3 -DWITH_GTEST_PARALLEL=OFF"
     fi
 
-    CMAKE_BUILD_OPTS="-DWITH_FIO=ON -DWITH_GTEST_PARALLEL=ON"
+    CMAKE_BUILD_OPTS="-DWITH_FIO=ON"
 
-    # Are we in the CI ?
+    cat <<EOM
+Note that the binaries produced by this script do not contain correct time
+and git version information, which may make them unsuitable for debugging
+and production use.
+EOM
+    save_ccache_conf
+    # remove the entropy generated by the date/time embedded in the build
+    CMAKE_BUILD_OPTS="$CMAKE_BUILD_OPTS -D ENABLE_GIT_VERSION=OFF"
```

Comment:
nit, could take this opportunity to remove the space after `-D` to be consistent with other part of this script.